### PR TITLE
trial refractor of errors

### PR DESCRIPTION
**Pull request by gnorbsl**

### CODE REVIEW

File: src/entities/customErrors/CustomError.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { logger } from '../../helper/logger';
```
Comment (Enubia) [inline src/entities/customErrors/CustomError.ts:1]: please move this file out of the entities directory, this one is only meant for entities in the system
create a folder like ```src/errors``` or something like that for it

File: src/entities/customErrors/InvalidInputError.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { CustomError } from './CustomError';
+import { CustomErrorCode, ICustomError } from '../../types/ICustomError';
+
+export class InvalidInputError extends CustomError {
+  public constructor(message?: string) {
```
Comment (Enubia) [inline src/entities/customErrors/InvalidInputError.ts:5]: please update all errors according to the custom error class, e.g add status and the error object to the constructor parameters, those are used for identifying status codes and error logging during development

**Review by Enubia — CHANGES_REQUESTED** [review]
Overall there are a lot of compile errors since the code prop is now an enum and I've used the custom error in a few places where a custom code is not possible anymore, please fix those errors as well.

Apart from that, looks good so far.

File: src/entities/customErrors/InvalidInputError.ts (hunk 2 above)
Comment (gnorbsl) [reply src/entities/customErrors/InvalidInputError.ts:5]: I'm not sure I understand what you mean, why would I need status and error here if they are optional anyway and not needed for this error.

**Pull request closed without merging**